infra/test/Test/Pos/Infra/Gen.hs

```diff
@@ -0,0 +1,110 @@
+                                  fromMicroseconds)
+
+import           Pos.Arbitrary.Core ()
+import           Pos.Core (EpochIndex (..), TimeDiff (..))
+import           Pos.Infra.Communication.Types.Relay (DataMsg (..),
+                                                      InvMsg (..),
+                                                      MempoolMsg (..),
+                                                      ReqMsg (..),
+                                                      ResMsg (..))
+import           Pos.Infra.DHT (DHTData (..), DHTKey (..), bytesToDHTKey)
+import           Pos.Infra.Slotting.Types (EpochSlottingData (..),
+                                           SlottingData,
+                                           createSlottingDataUnsafe)
+
+----------------------------------------------------------------------------
+-- DHT Generators
+----------------------------------------------------------------------------
+
+genInvMsg :: Gen a -> Gen (InvMsg a)
+genInvMsg genA = InvMsg <$> genA
+
+genReqMsg :: Gen (Maybe a) -> Gen (ReqMsg a)
+genReqMsg genMA = ReqMsg <$> genMA
+
+genResMsg :: Gen a -> Gen (ResMsg a)
+genResMsg genA = ResMsg <$> genA <*> Gen.bool
+
+genMempoolMsg :: Gen (MempoolMsg a)
+genMempoolMsg = return MempoolMsg
+
+genDataMsg :: Gen a -> Gen (DataMsg a)
+genDataMsg genA = DataMsg <$> genA
+
+genDHTKey :: Gen DHTKey
+genDHTKey = do
+    b <- gen32Bytes
+    let k = bytesToDHTKey b :: Either String DHTKey
+    case k of
+        Left _   -> error "Failed to generate a DHTKey."
+        Right dk -> return dk
+
+genDHTData :: Gen DHTData
+genDHTData = return $ DHTData ()
+
+----------------------------------------------------------------------------
+-- Slotting Generators
+----------------------------------------------------------------------------
+
+genEpochSlottingData :: Gen EpochSlottingData
+genEpochSlottingData = EpochSlottingData <$> genMillisecond <*> genTimeDiff
+
+genSlottingData :: Gen SlottingData
+genSlottingData = createSlottingDataUnsafe <$> genMap
+  where
+    genMap :: Gen (Map EpochIndex EpochSlottingData)
+    genMap = Gen.map Range.constantBounded genEpochIndexDataPair
+
+genEpochIndexDataPair :: Gen (EpochIndex, EpochSlottingData)
+genEpochIndexDataPair = do
+    i <- genEpochIndex
+    sd <- genEpochSlottingData
+    return (i, sd)
+
+----------------------------------------------------------------------------
+-- Helper Generators
+----------------------------------------------------------------------------
+
+genBytes :: Int -> Gen ByteString
+genBytes n = Gen.bytes (Range.singleton n)
+
+gen32Bytes :: Gen ByteString
+gen32Bytes = genBytes 32
+
+genMillisecond :: Gen Millisecond
+genMillisecond =
+    fromMicroseconds <$> (toInteger <$> Gen.int Range.constantBounded)
+
+genMicrosecond :: Gen Microsecond
+genMicrosecond =
+    fromMicroseconds <$> (toInteger <$> Gen.int Range.constantBounded)
+
+-- TimeDiff is from core so this generator will be moved eventually.
```

Comment:
@ruhatch Do you think I should wait until we get the `core` generators done before merging this PR? Since these generators will be included with the `core` gens.

Reply:
Yes and they should really be moved to `Test.Pos.Util.Gen` I think - push them right to the bottom of the package stack!